addons/InfoBar/InfoBar.lua

```diff
@@ -0,0 +1,303 @@
+defaults.TargetPC = "${name}"
+defaults.TargetNPC = "${name}"
+defaults.TargetMOB = "${name}"
+defaults.display = {}
+defaults.display.pos = {}
+defaults.display.pos.x = 0
+defaults.display.pos.y = 0
+defaults.display.bg = {}
+defaults.display.bg.red = 0
+defaults.display.bg.green = 0
+defaults.display.bg.blue = 0
+defaults.display.bg.alpha = 102
+defaults.display.text = {}
+defaults.display.text.font = 'Consolas'
+defaults.display.text.red = 255
+defaults.display.text.green = 255
+defaults.display.text.blue = 255
+defaults.display.text.alpha = 255
+defaults.display.text.size = 12
+
+settings = config.load(defaults)
+
+box = texts.new("", settings.display, settings)
+
+local infobar = {}
+infobar.new_line = '\n'
+
+windower.register_event('load',function()
+    db = sqlite3.open(windower.addon_path..'\database.db')
+    notesdb = sqlite3.open(windower.addon_path..'/notes.db')
+    notesdb:exec('CREATE TABLE IF NOT EXISTS notes(name TEXT primary key, note TEXT)')
+    if not windower.ffxi.get_info().logged_in then return end
+    local target = windower.ffxi.get_mob_by_target('st') or windower.ffxi.get_mob_by_target('t') or windower.ffxi.get_player()
+    get_target(target.index)
+end)
+
+windower.register_event('unload',function()
+    db:close()
+    notesdb:close()
+end)
+
+function getDegrees()
+    local target = windower.ffxi.get_mob_by_target('st') or windower.ffxi.get_mob_by_target('t') or windower.ffxi.get_mob_by_target('me')
+    local target_heading = V{}.from_radian(target.facing)
+    local angleInDegrees = (math.atan2(target_heading[1], target_heading[2]) * (180 / math.pi))
+    local Degrees = angleInDegrees % 360
+    return math.floor(Degrees)
+end
+
+function DegreesToDirection(Degrees)
+    local direction = (Degrees <= 11.25 and Degrees >= 0 and 'N') or
+        (Degrees <= 360 and Degrees > (11.25 * 31) and 'N') or
+        (Degrees <= (11.25 * 3) and Degrees > 11.25 and 'NNE') or
+        (Degrees <= (11.25 * 5) and Degrees > (11.25 * 3) and 'NE') or
+        (Degrees <= (11.25 * 7) and Degrees > (11.25 * 5) and 'NEE') or
+        (Degrees <= (11.25 * 9) and Degrees > (11.25 * 7) and 'E') or
+        (Degrees <= (11.25 * 11) and Degrees > (11.25 * 9) and 'SEE') or
+        (Degrees <= (11.25 * 13) and Degrees > (11.25 * 11) and 'SE') or
+        (Degrees <= (11.25 * 15) and Degrees > (11.25 * 13) and 'SSE') or
+        (Degrees <= (11.25 * 17) and Degrees > (11.25 * 15) and 'S') or
+        (Degrees <= (11.25 * 19) and Degrees > (11.25 * 17) and 'SSW') or
+        (Degrees <= (11.25 * 21) and Degrees > (11.25 * 19) and 'SW') or
+        (Degrees <= (11.25 * 23) and Degrees > (11.25 * 21) and 'SWW') or
+        (Degrees <= (11.25 * 25) and Degrees > (11.25 * 23) and 'W') or
+        (Degrees <= (11.25 * 27) and Degrees > (11.25 * 25) and 'NWW') or
+        (Degrees <= (11.25 * 29) and Degrees > (11.25 * 27) and 'NW') or
+        (Degrees <= (11.25 * 31) and Degrees > (11.25 * 29) and 'NNW')
+	return direction
+end
```

Comment:
This function does a lot of branches and calculations, it can be simplified thus:
```lua
local dir_sets = L{'E', 'ESE', 'SE', 'SSE', 'S', 'SSW', 'SW', 'WSW', 'W', 'WNW', 'NW', 'NNW', 'N', 'NNE', 'NE', 'ENE', 'E'}
function DegreesToDirection(val)
    return dir_sets[((val + 8)/16):floor() + 1]
end
```

Note that this function does not take the value in degrees, but in radians, i.e. just `target.facing`. See my comments on `getDegrees` above. As such you may want to rename the function accordingly (or just inline it in the calling function).

Reply:
I have a problem with this function so thats why I use the other one, with that function the facing is a little off, for example, when facing S (90º) it shows as SW.

Reply:
Hm, I just noticed this goes from 0 to 255, instead of proper radians... that's how it's transmitted in certain packets. To make it work for radians (which is what `target.facing` returns), change the scaling like this:
```lua
local dir_sets = L{'W', 'WNW', 'NW', 'NNW', 'N', 'NNE', 'NE', 'ENE', 'E', 'ESE', 'SE', 'SSE', 'S', 'SSW', 'SW', 'WSW', 'W'}
function DegreesToDirection(val)
    return dir_sets[math.round((val + math.pi) / math.pi * 8) + 1]
end
```

Reply:
thats returning nil now D:

Reply:
Are you sure you used the right value, the one returned by `mob.facing` as input?

Here's some sample code I wrote:
![image](https://user-images.githubusercontent.com/1635608/35571905-a6dd523e-05d3-11e8-8696-f0b894b4295d.png)

And here is the result:
![dir](https://user-images.githubusercontent.com/1635608/35571941-c1d86d80-05d3-11e8-8ce7-e7b72f7daacb.gif)

(`target.me.facing` up there is the same as `windower.ffxi.get_mob_by_target('me').facing`.)

Reply:
sorry, it was an error on my end, will update the pull request soon.